L1Trigger/TrackerTFP/src/MiniHoughTransform.cc

```diff
@@ -0,0 +1,285 @@
+#include "L1Trigger/TrackerTFP/interface/MiniHoughTransform.h"
+
+#include <numeric>
+#include <algorithm>
+#include <iterator>
+#include <deque>
+#include <vector>
+#include <set>
+#include <utility>
+#include <cmath>
+
+using namespace std;
+using namespace edm;
+using namespace trackerDTC;
+
+namespace trackerTFP {
+
+  MiniHoughTransform::MiniHoughTransform(const ParameterSet& iConfig, const Setup* setup, const DataFormats* dataFormats, int region) :
+    enableTruncation_(iConfig.getParameter<bool>("EnableTruncation")),
+    setup_(setup),
+    dataFormats_(dataFormats),
+    inv2R_(dataFormats_->format(Variable::inv2R, Process::ht)),
+    phiT_(dataFormats_->format(Variable::phiT, Process::ht)),
+    region_(region),
+    numBinsInv2R_(setup_->htNumBinsInv2R()),
+    numCells_(setup_->mhtNumCells()),
+    numNodes_(setup_->mhtNumDLBNodes()),
+    numChannel_(setup_->mhtNumDLBChannel()),
+    input_(numBinsInv2R_) {}
+
+  void MiniHoughTransform::consume(const TTDTC::Streams& streams) {
+    auto valid = [](int& sum, const TTDTC::Frame& frame){ return sum += (frame.first.isNonnull() ? 1 : 0); };
```

Comment:
Add comment "Read in & organise input data (filling array input_)"

Reply:
done